### PR TITLE
Deliverable 2 Added Hooks and Context

### PR DESCRIPTION
**Pull request by alma1997alvarez**
## Description
**Mini-Challenge 3: Hooks:**

- Fetched videos from the Youtube API.
- Added functionality to the Searchbar component.
- Added the VideoDetailsView component to display the details of any video that is clicked on.
- Retrieved related videos from the Youtube API for the active one.

**Mini-Challenge 4: State Management**

- Created the Provider for the Global Context.
- Consumed the Global Context inside the components that require it.
- Themes stored globally and is changed based on the ToggleDarkMode component choice.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking refactor which doesn't change functionality)
- [x] Styles (Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc))

## How should this be tested?
Implemented tests using Jest and react testing library. To test:

`npm run test`

## What is the deliverable?

Deliverable 2: Mini challenges 3 and 4

## Screenshots and links
Live demo [here](https://61e5c76810fde32c30a70508--alma1997alvarez-react-apprenticeship.netlify.app/).

Tests coverage:
Screen Shot 2022-01-17 at 17 22 59

## Notes

Thanks for all the help and practice with desctructuring assignments, John! 🌟 

## Checklist
- [x] I added the necessary documentation, if appropriate.
- [x] I added tests to prove that my fix is effective or my deliverable works.
- [x] I reviewed existing Pull Requests before submitting mine.
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/App/App.component.jsx [hunk 1]
```diff
@@ -1,22 +1,57 @@
...
-import MainHeader from '../Header';
+import Header from '../Header';
 import Heading from '../Heading';
 import MainContainer from '../MainContainer';
 import VideosContainer from '../VideosContainer';
+import VideoDetailsView from '../VideoDetailsView';
 
 function App() {
+  const theme = useContext(ThemeContext);
+  const darkMode = theme.state.darkMode;
```
Comment (juanhdezran88) [inline src/components/App/App.component.jsx]: Destructure darkMode so we can access it in this way 
`const { state: { darkMode = false } = {} } = theme;`

File: src/components/DarkModeToggle/DarkModeToggle.component.jsx [hunk 2]
```diff
@@ -9,15 +10,32 @@ import {
 } from './DarkModeToggle.styled';
 
 const DarkModeToggle = () => {
+  const theme = useContext(ThemeContext);
+  const darkMode = theme.state.darkMode;
```
Comment (juanhdezran88) [inline src/components/DarkModeToggle/DarkModeToggle.component.jsx]: Same as the comment above, destructure darkMode

File: src/components/DarkModeToggle/DarkModeToggle.component.jsx [hunk 3]
```diff
@@ -9,15 +10,32 @@ import {
...
+
+  const onClickHandler = () => {
+    if (darkMode) {
+      theme.dispatch({ type: 'LIGHTMODE' });
```
Comment (juanhdezran88) [inline src/components/DarkModeToggle/DarkModeToggle.component.jsx]: Also you can destructure `dispatch` in the same line of `darkMode`

File: src/components/VideoCard/VideoCard.component.jsx [hunk 4]
```diff
@@ -8,11 +9,27 @@ import {
 } from './VideoCard.styled';
 
 const VideoCard = (props) => {
+  const currentVideoContext = React.useContext(CurrentVideoContext);
+  const onClickHandler = () => {
+    currentVideoContext.setVideoDetails({
+      id: props.id,
```
Comment (juanhdezran88) [inline src/components/VideoCard/VideoCard.component.jsx]: it's a good practice to destructure the props component and use them the single word
`const { id = '', title = '', description = ''} = props`

or destructure them in the params
`const VideCard = ({ id = '', title = '', description = ''} ) => {`

File: src/components/VideoDetailsCard/VideoDetailsCard.component.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  VideoDetailsIframe,
+  VideoDetailsCardStyled,
+} from './VideoDetailsCard.styled';
+
+const VideoDetailsCard = () => {
+  const currentVideoContext = useContext(CurrentVideoContext);
+  return (
+    <VideoDetailsCardStyled>
+      <VideoDetailsIframe
+        src={`https://www.youtube.com/embed/${currentVideoContext.videoDetails.id}`}
```
Comment (juanhdezran88) [inline src/components/VideoDetailsCard/VideoDetailsCard.component.js]: will be a good practice to destructure all the videoDetails properties

File: src/hooks/usePopulateVideoCards.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import VideoCard from '../components/VideoCard';
+
+const usePopulateVideoCards = (videos, displayDescription = true) => {
+  let videoCardsPopulated = <></>;
+  const { items } = videos;
```
Comment (juanhdezran88) [inline src/hooks/usePopulateVideoCards.js]: maybe we can add a default value

File: src/components/App/App.component.jsx (hunk 1 above)
Comment (alma1997alvarez) [reply src/components/App/App.component.jsx]: Done! ⭐

File: src/components/DarkModeToggle/DarkModeToggle.component.jsx (hunk 3 above)
Comment (alma1997alvarez) [reply src/components/DarkModeToggle/DarkModeToggle.component.jsx]: Also done! ⭐

File: src/components/VideoCard/VideoCard.component.jsx (hunk 4 above)
Comment (alma1997alvarez) [reply src/components/VideoCard/VideoCard.component.jsx]: Went with the first option!

File: src/components/VideoDetailsCard/VideoDetailsCard.component.js (hunk 5 above)
Comment (alma1997alvarez) [reply src/components/VideoDetailsCard/VideoDetailsCard.component.js]: Done! ⭐

File: src/hooks/usePopulateVideoCards.js (hunk 6 above)
Comment (alma1997alvarez) [reply src/hooks/usePopulateVideoCards.js]: Yes! Just did that. Thank you!

**Review by juanhdezran88 — APPROVED** [review]

**Pull request merged**